File: 23. Merge k Sorted Lists/merge_k_sorted_lists.py
class ListNode(object):
     def __init__(self, val=0, next=None):
         self.val = val
         self.next = next

class Solution(object):
    def mergeKLists(self, lists):
        """
        :type lists: List[ListNode]
        :rtype: ListNode
        """
        sorted_list = []

        for head in lists:
            while head:
                sorted_list.append(head.val)
                head = head.next

        sorted_list = sorted(sorted_list)
        dummy = ListNode(0)
        current = dummy

        for value in sorted_list:
            current.next = ListNode(value)
            current = current.next

        return dummy.next
        

File: 23. Merge k Sorted Lists/MergeKSortedLists.cs
public class Solution {
    public ListNode MergeKLists(ListNode[] lists) {
        var sortedList = new List<int>();

        foreach (var head in lists) {
            var currentHeadNode = head;
            while (currentHeadNode != null) {
                sortedList.Add(currentHeadNode.val);
                currentHeadNode = currentHeadNode.next;
            }
        }

        sortedList.Sort();

        var dummyNode = new ListNode(0);
        var currentNode = dummyNode;

        foreach (var value in sortedList) {
            currentNode.next = new ListNode(value);
            currentNode = currentNode.next;
        }

        return dummyNode.next;
    }
}